TFNetworkRecLayer.py

```diff
@@ -3346,23 +3346,33 @@ def __init__(self, beam_size, input_type="prob", explicit_search_source=None, le
         scores_in = tf.log(scores_in)
       elif input_type == "log_prob":
         pass
+      elif input_type == "regression":
+        pass
       else:
         raise Exception("%r: invalid input type %r" % (self, input_type))
-      samples = tf.multinomial(scores_in, num_samples=1, seed=get_random_seed())  # (batch, num_samples), int64
-      samples = tf.to_int32(tf.reshape(samples, [-1]))  # (batch,), int32
-      if self.scheduled_sampling.get("gold_mixin_prob"):
+
+      if input_type == "regression":
+        feedback_output = scores_in
+      else:
+        # sample from scores
+        samples = tf.multinomial(scores_in, num_samples=1, seed=get_random_seed())  # (batch, num_samples), int64
+        feedback_output = tf.to_int32(tf.reshape(samples, [-1]))  # (batch,), int32
+
+      gold_mixing_prob = self.scheduled_sampling.get("gold_mixin_prob", False)
+      if gold_mixing_prob:
         gold_targets = self._get_target_value().get_placeholder_as_batch_major()  # (batch,), int32
-        samples = tf.where(
-          tf.less(tf.random_uniform(tf.shape(samples)), self.scheduled_sampling.get("gold_mixin_prob")),
-          gold_targets, samples)
+        choice_shape = tf.shape(feedback_output) if input_type is not "regression" else (tf.shape(feedback_output)[0],)
```

Comment:
`is not` is wrong. Also, why not just `tf.shape(scores_in)[:-1]`. It's simpler and shorter. Also I would not introduce the local var for it but just have it in `choice = ...` directly.